prov/mrail/src/mrail.h

```diff
@@ -353,3 +353,17 @@ void mrail_free_req(struct mrail_ep *mrail_ep, struct mrail_req *req)
 void mrail_progress_deferred_reqs(struct mrail_ep *mrail_ep);
 
 void mrail_poll_cq(struct util_cq *cq);
+
+static inline void mrail_cntr_inc(struct util_cntr *cntr)
+{
+       if (cntr) {
+               cntr->cntr_fid.ops->add(&cntr->cntr_fid, 1);
+       }
+}
+
+static inline void mrail_cntr_incerr(struct util_cntr *cntr)
+{
+       if (cntr) {
+               cntr->cntr_fid.ops->adderr(&cntr->cntr_fid, 1);
+       }
+}
```

Comment:
For success counters, I would use function pointers to increment functions, with a no-op version to handle the case where no counter has been assigned.  The error counter could be different and continue to have a check, so we can avoid needing function pointers for unexpected cases.

Reply:
I just pushed some changes to add the function pointers within the util_ep. Is that how you envisioned it?

Reply:
Yes, thanks